gh-46236: Add missing document of `PyUnicode_DecodeCodePageStateful` (#127934)
python/cpython: Doc/c-api/unicode.rst

   in *consumed*.

.. c:function:: PyObject* PyUnicode_DecodeCodePageStateful(int code_page, const char *str, \
                              Py_ssize_t size, const char *errors, Py_ssize_t *consumed)

>    Same as :c:func:`PyUnicode_DecodeMBCSStateful`, except that using the code page

Reviewer: ```suggestion
   Similar to :c:func:`PyUnicode_DecodeMBCSStateful`, except that using the code page
```

Why use same just to contradict it?

>    Similar to :c:func:`PyUnicode_DecodeMBCSStateful`, except that using the code page

Reviewer 2: ```suggestion
   Similar to :c:func:`PyUnicode_DecodeMBCSStateful`, except uses the code page
```

>    specified by *code_page*.  Use :c:macro:`!CP_ACP` code page to get the MBCS decoder.

Reviewer 2: I don't see the need for this part. If you're using `PyUnicode_DecodeCodePageStateful`, you don't care about the MBCS decoder, because otherwise you would be using `PyUnicode_DecodeMBCSStateful`, right?
```suggestion
   specified by *code_page*.
```
